### PR TITLE
UMD main branch for TT-Metal

### PR DESCRIPTION
**Pull request by vtangTT**
~~TT-Metal will point to this branch as we make changes unrelated to BBE~~

Current changes:

- Deleted all dead code
- moved versim and emulation in `simulation/deprecated` folder

metal post-commit: https://github.com/tenstorrent/tt-metal/actions/runs/10794632975

commit history can be found here: https://github.com/tenstorrent/tt-umd/tree/vtang/metal-main-history

### CODE REVIEW

**Comment by broskoTT** [comment]
I see that versim device is completely deleted. Is this something we want to still maintain? @joelsmithTT @ttmtrajkovic

File: device/simulation/tt_simulation_device.h [hunk 1]
```diff
@@ -57,24 +55,20 @@ class tt_SimulationDevice: public tt_device {
     // Misc. Functions to Query/Set Device State
     // virtual bool using_harvested_soc_descriptors();
     virtual std::unordered_map<chip_id_t, uint32_t> get_harvesting_masks_for_soc_descriptors();
-    // virtual bool noc_translation_en();
     // virtual void translate_to_noc_table_coords(chip_id_t device_id, std::size_t &r, std::size_t &c);
```
Comment (pjanevskiTT) [inline device/simulation/tt_simulation_device.h:58]: Why are these functions under comment? Are these functions used anywhere?

Comment (vtangTT) [reply device/simulation/tt_simulation_device.h:58]: Ah no they're not used. Originally copy pasted from `tt_silicon_driver` and then commented out functions that I didn't need for the simulation but then forgot about them

I'll delete these commens

File: device/tt_cluster_descriptor.cpp [hunk 2]
```diff
@@ -52,19 +52,6 @@ std::vector<std::tuple<ethernet_channel_t, ethernet_channel_t>> tt_ClusterDescri
...
-
-    if (this->ethernet_connections.at(first).find(first_channel) == this->ethernet_connections.at(first).end()) {
-        return false;
-    }
-
-    const auto &[connected_chip, connected_channel] = this->ethernet_connections.at(first).at(first_channel);
-    return connected_chip == second && connected_channel == second_channel;   
-}
-
 // const eth_coord_t tt_ClusterDescriptor::get_chip_xy(const chip_id_t &chip_id) const {
```
Comment (pjanevskiTT) [inline device/tt_cluster_descriptor.cpp:55]: Same as above, if this is dead code we can delete it

File: device/tt_cluster_descriptor.h [hunk 3]
```diff
@@ -87,7 +86,6 @@ class tt_ClusterDescriptor {
   // const eth_coord_t get_chip_xy(const chip_id_t &chip_id) const;
```
Comment (pjanevskiTT) [inline device/tt_cluster_descriptor.h:86]: Same as above

**Review by pjanevskiTT — APPROVED** [review]

**Review by joelsmithTT — APPROVED** [review]

File: device/tt_cluster_descriptor.h (hunk 3 above)
Comment (vtangTT) [reply device/tt_cluster_descriptor.h]: deleted

File: device/tt_cluster_descriptor.cpp (hunk 2 above)
Comment (vtangTT) [reply device/tt_cluster_descriptor.cpp]: deleted

File: device/CMakeLists.txt [hunk 4]
```diff
@@ -1,20 +1,18 @@
...
     cpuset_lib.cpp
-    grayskull_implementation.cpp
     tlb.cpp
     tt_cluster_descriptor.cpp
     tt_device.cpp
-    tt_emulation_stub.cpp
     tt_silicon_driver.cpp
     tt_silicon_driver_common.cpp
     tt_soc_descriptor.cpp
-    tt_versim_stub.cpp
```
Comment (broskoTT) [inline device/CMakeLists.txt:14]: Is this intentional? Along with tt_emulation_stub.cpp?

**Review by broskoTT — APPROVED** [review]

File: device/CMakeLists.txt (hunk 4 above)
Comment (vtangTT) [reply device/CMakeLists.txt:14]: Yes, they're not functional so will move to a deprecated folder for now.

**Pull request merged**